src/index.js

```diff
@@ -119,10 +119,14 @@ class Dropzone extends React.Component {
     }
   }
 
-  onClick() {
+  onClick(e) {
     if (!this.props.disableClick) {
       this.open();
     }
+
+    if (this.props.onClick) {
+      this.props.onClick.call(this, e);
```

Comment:
Probably. Leave it for now. This needs to be refactored for all places. FIY propTypes does not guard in runtime, that's why I think it's a good idea to check the type.